_posts/python/2016-07-26-get-slope-aspect-from-digital-elevation-model.md

```diff
@@ -22,162 +22,83 @@ With this data, we are able to use programs in order to analyze certain physical
-    Check the NumPy 1.11 release notes for more information.
-      self.data[data.mask] = FILL_VALUE
-
+## Load the DEM with richdem
 
-    starting slope/direction calculation for chunk 1 [0:516, 0:516]
-    starting slope/direction calculation for chunk 2 [0:516, 508:1028]
-    starting slope/direction calculation for chunk 3 [0:516, 1020:1080]
-    starting slope/direction calculation for chunk 4 [508:1028, 0:516]
-    starting slope/direction calculation for chunk 5 [508:1028, 508:1028]
-    starting slope/direction calculation for chunk 6 [508:1028, 1020:1080]
-    starting slope/direction calculation for chunk 7 [1020:1080, 0:516]
-    starting slope/direction calculation for chunk 8 [1020:1080, 508:1028]
-    starting slope/direction calculation for chunk 9 [1020:1080, 1020:1080]
-
-
-Let's peek at these objects, both of which are numeric arrays. 
+The `richdem` package has a `LoadGDAL` function for loading DEMs that we can use here. 
 
 
 ```python
-mag
+shasta_dem = rd.LoadGDAL("Shasta-30m-DEM.tif")
 ```
 
 
-
-
-    array([[        nan,         nan,         nan, ...,  0.32364485,
-             0.44950795,  0.2586934 ],
-           [        nan,         nan,         nan, ...,  0.32364485,
-             0.44950795,  0.41471453],
-           [        nan,         nan,  0.04311538, ...,  0.43404787,
-             0.4197514 ,  0.28500046],
-           ..., 
-           [ 0.44798625,  0.44798625,  0.51895544, ...,  0.1062964 ,
-             0.12971104,  0.12971104],
-           [ 0.3983211 ,  0.47653369,  0.52412218, ...,  0.07775003,
-             0.10756881,  0.1297108 ],
-           [ 0.42918458,  0.5150215 ,  0.55793996, ...,  0.07775003,
-             0.10756881,  0.08583692]])
-
-
-
-
 ```python
-aspect
+plt.imshow(shasta_dem, interpolation='none')
+plt.colorbar()
+plt.show()
 ```
 
 
+![png](/images/get-slope-aspect-from-digital-elevation-model_files/get-slope-aspect-from-digital-elevation-model_7_0.png)
 
 
-    array([[        nan,         nan,         nan, ...,  2.49694673,
-             2.6124799 ,  3.14159265],
-           [        nan,         nan,         nan, ...,  2.49694673,
-             2.6124799 ,  2.24437453],
-           [        nan,         nan,  3.14159265, ...,  2.67698894,
-             2.37318076,  2.22072197],
-           ..., 
-           [ 0.53113069,  0.53113069,  0.59700458, ...,  4.29676177,
-             3.98922813,  3.98922813],
-           [ 0.2466342 ,  6.14671481,  6.09656191, ...,  4.12762132,
-             3.78846596,  3.98923025],
-           [ 0.        ,  0.        ,  0.        , ...,  4.12762132,
-             3.78846596,  3.14159265]])
-
+## Compute and plot slope and aspect
 
+We can use the `rd.TerrainAttribute` function to compute slope and aspect for each pixel. Note that there are multiple ways to represent the slope values. See https://richdem.readthedocs.io/en/latest/python_api.html#richdem.TerrainAttribute for more options.
```

Comment:
You will need to write out the url and ideally have it open in a new window. Can you try this:
```
<a href="https://richdem.readthedocs.io/en/latest/python_api.html#richdem.TerrainAttribute" target="_blank">Read the richdem docs for more options.</a>
```

Reply:
Ah cool thanks for writing it out!